File: module-core/src/main/java/maple/expectation/error/exception/auth/CharacterNotOwnedException.java
package maple.expectation.error.exception.auth;

import maple.expectation.error.CommonErrorCode;
import maple.expectation.error.exception.base.ClientBaseException;
import maple.expectation.error.exception.marker.CircuitBreakerIgnoreMarker;

/** 캐릭터 소유권 검증 실패 예외 */
public class CharacterNotOwnedException extends ClientBaseException
    implements CircuitBreakerIgnoreMarker {

  public CharacterNotOwnedException(String userIgn) {
    super(CommonErrorCode.CHARACTER_NOT_OWNED, userIgn);
  }
}

File: module-common/src/main/kotlin/maple/expectation/error/exception/CharacterNotOwnedException.kt
package maple.expectation.error.exception

import maple.expectation.error.CommonErrorCode
import maple.expectation.error.exception.base.ClientBaseException

class CharacterNotOwnedException : ClientBaseException {

    /**
     * Constructor with user IGN or character OCID
     */
    constructor(identifier: String) : super(
        CommonErrorCode.CHARACTER_NOT_OWNED,
        identifier
    )
}
